mutation_partition.cc

```diff
@@ -1290,17 +1291,20 @@ void mutation_partition::trim_rows(const schema& s,
 }
 
 uint32_t mutation_partition::do_compact(const schema& s,
+    const dht::decorated_key& dk,
     gc_clock::time_point query_time,
     const std::vector<query::clustering_range>& row_ranges,
     bool always_return_static_content,
     bool reverse,
     uint64_t row_limit,
-    can_gc_fn& can_gc)
+    can_gc_fn& can_gc,
+    bool drop_tombstones_unconditionally)
 {
     check_schema(s);
     assert(row_limit > 0);
 
-    auto gc_before = saturating_subtract(query_time, s.gc_grace_seconds());
+    auto gc_before = drop_tombstones_unconditionally ? gc_clock::time_point::max() :
+        ::get_gc_before_for_key(s.shared_from_this(), dk, query_time);
```

Comment:
isn't it better to have do_compact() caller to pass gc_before instead?

Reply:
we have multiple places to call do_compact(), it is better to calculate the gc_before in a single place, no?